pkg/yurtmanager/controller/loadbalancerset/loadbalancerset/map_utils.go

```diff
@@ -0,0 +1,154 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an AS IS BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package loadbalancerset
+
+import (
+	"sort"
+	"strings"
+)
+
+const (
+	aggregateKeyPrefix = "aggregated."
+)
+
+func mergeMap(m map[string]string, elem map[string]string) map[string]string {
+	if len(elem) == 0 {
+		return m
+	}
+
+	if m == nil {
+		m = make(map[string]string)
+	}
+
+	for k, v := range elem {
+		m[k] = mergeMapValue(m[k], v)
+	}
+
+	return m
+}
+
+func mergeMapValue(originalValue, addValue string) string {
+	if len(originalValue) == 0 {
+		return addValue
+	}
+
+	if len(addValue) == 0 {
+		return originalValue
+	}
+
+	splitOriginalValues := strings.Split(originalValue, ",")
+	if valueIsExist(splitOriginalValues, addValue) {
+		return originalValue
+	}
+
+	return joinNewValue(splitOriginalValues, addValue)
+}
+
+func valueIsExist(originalValueList []string, addValue string) bool {
+	for _, oldValue := range originalValueList {
+		if addValue == oldValue {
+			return true
+		}
+	}
+	return false
+}
+
+func joinNewValue(originalValueList []string, addValue string) string {
+	originalValueList = append(originalValueList, addValue)
+	sort.Strings(originalValueList)
+
+	return strings.Join(originalValueList, ",")
+}
+
+func addAggregatePrefix(m map[string]string) map[string]string {
+	if m == nil {
+		return nil
+	}
+
+	newMap := make(map[string]string)
+	for key, value := range m {
+		newMap[aggregateKeyPrefix+key] = value
+	}
+
+	return newMap
+}
+
+func filterIgnoredKeys(m map[string]string) map[string]string {
+	newMap := make(map[string]string)
+	for key, value := range m {
+		if !strings.HasPrefix(key, aggregateKeyPrefix) {
+			continue
+		}
+		newMap[key] = value
+	}
+	return newMap
+}
+
+func diffMap(currentMap, desiredMap map[string]string) (update map[string]string, deletion map[string]string) {
+	if currentMap == nil {
+		return desiredMap, nil
+	}
+	if desiredMap == nil {
+		return nil, currentMap
+	}
+
+	update = make(map[string]string)
+	for key, value := range desiredMap {
+		if currentMap[key] != value {
+			update[key] = value
+		}
+	}
+
+	deletion = make(map[string]string)
+	for key, value := range currentMap {
+		if _, exist := desiredMap[key]; !exist {
+			deletion[key] = value
+		}
+	}
+	return
+}
+
+func updateMap(m, update, deletion map[string]string) map[string]string {
+	if len(update) == 0 && len(deletion) == 0 {
+		return m
+	}
+	if m == nil {
+		m = make(map[string]string)
+	}
+	for key, value := range update {
+		m[key] = value
+	}
+
+	for key := range deletion {
+		delete(m, key)
+	}
+	return m
+}
```

Comment:
diffMap and updateMap can be optimized as one function:

```
func mergeDesiredMap(currentMap, desiredMap map[string]string) bool {
    if currentMap == nil {
        currentMap = make(map[string]string)
    }

    modified := false
    for key, value := range desiredMap {
        if currentValue, exists := currentMap[key]; !exists || currentValue != value {
            currentMap[key] = value
            modified = true
        }
    }

    for key := range currentMap {
        if _, exists := desiredMap[key]; !exists {
            delete(currentMap, key)
            modified = true
        }
    }

    return modified
}
```

Reply:
done